src/java/fr/paris/lutece/plugins/appointment/modules/ants/web/AppointmentAnts.java

```diff
@@ -34,92 +34,143 @@
+	 * @throws AccessDeniedException
+	 */
+	@View(value = VIEW_PREDEMANDEFORM, defaultView = true)
+	public XPage viewPreDemandeForm(HttpServletRequest request) {
+		String dateTime = request.getParameter(PARAMETER_DATE_TIME);
+		String idForm = request.getParameter(PARAMETER_PROPERTY_ID_FORM);
+		String nbPlacesToTake = request.getParameter(PARAMETER_PLACES_TAKED_NOMBER_VALUE);
+
+		Map<String, Object> model = getModel();
+		model.put(MARKER_STRATING_DATE_TIME, dateTime);
+		model.put(MARKER_ID_FORM, idForm);
+		model.put(MARKER_NB_PLACES_TO_TAKE, nbPlacesToTake);
+
+		return getXPage(TEMPLATE_PREDEMANDEFORM, request.getLocale(), model);
+	}
+
+	/**
+	 * Redirects to main carto pview
+	 *
+	 * @param request The HTTP request
+	 * @return The view
+	 */
+	@Action(value = ACTION_PRE_SEARCH)
+	public XPage presearch(HttpServletRequest request) throws IOException {
+		Integer nbPlacesToTake = Integer.parseInt(request.getParameter(PARAMETER_PLACES_TAKED_NOMBER_VALUE));
+		String fieldsErrorMessage = AppPropertiesService.getProperty(PROPERTY_ERROR_MESSAGE);
+		String dateTime = request.getParameter(PARAMETER_DATE_TIME);
+
+		List<String> predemandeCodeList = PredemandeCodeUtils.getPredemandeCodeList(request,
+				PROPERTY_ID_PREDEMANDE_CODE_SUFFIX, nbPlacesToTake);
+
+		PredemandeCodeUtils.insertCodesPredemandeOnSession(request, predemandeCodeList);
+
+		XPage redirectionXpage = new XPage();
+		String url = null;
+
+		boolean isAllCodesNotValid = PreDemandeValidationService.processPreDemandeCodes(predemandeCodeList);
+
+		if (StringUtils.isNotBlank(dateTime)) {
+			if (!isAllCodesNotValid) {
+				addError(fieldsErrorMessage);
+				url = PredemandeCodeUtils.constructRedirectionUrl(request, predemandeCodeList, null, null,
+						XPAGE_NAME, VIEW_PREDEMANDEFORM, PARAMETER_PROPERTY_ID_FORM, PARAMETER_DATE_TIME, dateTime,
+						PARAMETER_PLACES_TAKED_NOMBER, PARAMETER_PLACES_TAKED_NOMBER_VALUE, null, null);
+				redirectionXpage = redirect(request, url);
+			} else {
+				url = PredemandeCodeUtils.constructRedirectionUrl(request, predemandeCodeList, null, null,
+						APPOINTMENT_PLUGIN_XPAGE_NAME, APPOINTMENT_PLUGIN_APPOINTMENTFORM_VIEW_NAME,
+						PARAMETER_PROPERTY_ID_FORM, PARAMETER_DATE_TIME, dateTime, PARAMETER_PLACES_TAKED_NOMBER,
+						PARAMETER_PLACES_TAKED_NOMBER_VALUE, PARAMETER_ANCHOR, PARAMETER_ANCHOR_VALUE);
+				redirectionXpage = redirect(request, url);
+			}
+
+		} else {
+			if (!isAllCodesNotValid) {
+				addError(fieldsErrorMessage);
+				redirectionXpage = redirectView(request, VIEW_PREDEMANDEFORM);
+			} else {
+				url = PredemandeCodeUtils.constructRedirectionUrl(request, predemandeCodeList, PARAMETER_CATEGORIE,
+						PARAMETER_CATEGORIE_TITRES, APPOINTMENTSEARCH_PLUGIN_XPAGE_NAME,
+						APPOINTMENTSEARCH_PLUGIN_SEARCH_VIEW_NAME, null, null, null, PARAMETER_NB_CONSECUTIVE_SLOTS,
+						PARAMETER_PLACES_TAKED_NOMBER_VALUE, null, null);
+				redirectionXpage = redirect(request, url);
+			}
+		}
+
+		return redirectionXpage;
```

Comment:
all this code should be simply :  first the error cases, and then the normal case,
for example (i don't know if those are the correct VIews to redirect) : 

```

Map additionalParameters = PredemandeCodeUtils.buildAttributesMap( ...) ;  // if "startingDateTime" is empty, it will be handled in this method

if (!isAllCodesValid) {
	addError(  MESSAGE_EXISTS_ONE_INVALID_CODE_AT_LEAST );	
	redirect( request, APPOINTMENT_PLUGIN_APPOINTMENTFORM_VIEW_NAME, additionalParameters );
}

return redirect( request, APPOINTMENTSEARCH_PLUGIN_XPAGE_NAME, additionalParameters );
				

```

Reply:
Thanks for your remark ! 

We have 4 distinct redirection scenarios in our application, each with its own XPage and view name requirements. Dynamic determination of XPage and view based on conditions is essential to accurately meet these different scenarios.
I have tried to use the redirect method with additionalParameters, but the `getViewUrl(strView)` is responsible for determining the XPage name for the redirect. This line retrieves the XPage URL based on the strView parameter, which is supplied as an argument to the redirect method. After using this solution, I obtained something like this: 
`/jsp/site/Portal.jsp?page=appointmentants&view=Portal.jsp?page=appointment&view=getViewAppointmentForm&starting_date_time=2023-09-26T19:00&id_form=1&anchor=#step3&nbPlacesToTake=1`